src/GraphQL/Execution/ErrorInfoProvider.cs

```diff
@@ -65,14 +69,32 @@ public virtual ErrorInfo GetInfo(ExecutionError executionError)
                         extensions.Add("number", number);
                     if (data != null)
                         extensions.Add("data", data);
+                    if (stacktrace != null)
+                        extensions.Add("stacktrace", stacktrace);
                 }
             }
 
             return new ErrorInfo
             {
-                Message = _options.ExposeExceptionStackTrace ? executionError.ToString() : executionError.Message,
+                Message = _options.ExposeExceptionStackTrace && _options.ExposeExceptionStackTraceMode == ExposeExceptionStackTraceMode.Message ? executionError.ToString() : executionError.Message,
                 Extensions = extensions,
             };
+
+            string BuildStackTrace(Exception ex)
+            {
+                string prefix = "";
+                StringBuilder sb = new();
+                while (ex != null)
+                {
+                    AppendEntry();
+                    ex = ex.InnerException!;
+                    prefix = "[inner] ";
+                }
+
+                return sb.ToString();
+
+                void AppendEntry() => sb.Append(prefix).Append(ex.GetType().FullName).AppendLine(":").AppendLine(ex.StackTrace ?? "   <no stacktrace>");
+            }
```

Comment:
This basically just replicates `ToString()` in a different manner and without the messages of exceptions (outer or inner exceptions).  I would just remove and replace with `ToString()`.

```suggestion
```

Reply:
Good. I would too. I just wanted to confirm this intent.